### PR TITLE
feat(APIv2): RHINENG-8773 introduce the rule results endpoint

### PR DESCRIPTION
**Pull request by skateman**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: app/policies/v2/rule_result_policy.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    end
+
+    def match_group?
+      groups = user.inventory_groups
+      system = record.system
+      # Global access || ungrouped host || group matching
+      (groups == Rbac::ANY) || (system.groups.blank? && groups&.include?([])) || system.group_ids.intersect?(groups)
+    end
+
+    # Only show test results in our user account
```
Comment (romanblanco) [inline app/policies/v2/rule_result_policy.rb:27]:
```suggestion
    # Only show rule results in our user account
```

File: app/serializers/v2/rule_result_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    derived_attribute :system_id, system: [:id] # This is for the policy to work properly
+
+    derived_attribute :ref_id, rule: [:ref_id]
+    derived_attribute :title, rule: [:title]
+    derived_attribute :rationale, rule: [:rationale]
+    derived_attribute :description, rule: [:description]
+    derived_attribute :severity, rule: [:severity]
+    derived_attribute :precedence, rule: [:precedence]
+    derived_attribute :identifier, rule: [:identifier]
+    derived_attribute :references, rule: [:references]
```
Comment (romanblanco) [inline app/serializers/v2/rule_result_serializer.rb:17]:
```suggestion
    derived_attribute :references, rule: [:references]
    derived_attribute :value_checks, rule: [:value_checks]
```

File: spec/controllers/v2/rule_results_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      rule_id: :rule_id,
+      system_id: :system_id,
+      ref_id: :ref_id,
+      title: :title,
+      rationale: :rationale,
+      description: :description,
+      severity: :severity,
+      precedence: :precedence,
+      identifier: :identifier,
+      references: :references,
```
Comment (romanblanco) [inline spec/controllers/v2/rule_results_controller_spec.rb:18]:
```suggestion
      references: :references,
      value_checks: :value_checks,
```

File: app/serializers/v2/rule_result_serializer.rb (hunk 2 above)
Comment (skateman) [reply app/serializers/v2/rule_result_serializer.rb:17]: I left this field out on purpose, it makes no sense here.

**Review by romanblanco — APPROVED** [review]
LGTM :+1:

**Pull request merged**